File: Swift-C++-IR-Comparison/test/structsAndClasses/Swift/Classes/Deinitializer.swift
@main
struct MyMain {

    static func main() {
        var playerOne: Player? = Player(coins: 100)
        playerOne!.win(coins: 2_000)
        print("PlayerOne won 2000 coins & now has \(playerOne!.coinsInPurse) coins")
        // Prints "PlayerOne won 2000 coins & now has 2100 coins"
        print("The bank now only has \(Bank.coinsInBank) coins left")
        playerOne = nil
        print("PlayerOne has left the game")
        // Prints "PlayerOne has left the game"
        print("The bank now has \(Bank.coinsInBank) coins")
    }
}

class Bank {
    static var coinsInBank = 10_000
    static func distribute(coins numberOfCoinsRequested: Int) -> Int {
        let numberOfCoinsToVend = min(numberOfCoinsRequested, coinsInBank)
        coinsInBank -= numberOfCoinsToVend
        return numberOfCoinsToVend
    }
    static func receive(coins: Int) {
        coinsInBank += coins
    }
}

class Player {
    var coinsInPurse: Int
    init(coins: Int) {
        coinsInPurse = Bank.distribute(coins: coins)
    }
    func win(coins: Int) {
        coinsInPurse += Bank.distribute(coins: coins)
    }
    deinit {
        Bank.receive(coins: coinsInPurse)
    }
}

File: AutoTranspiledTests/try-1/Deinitializer.cpp

 #include<iostream>
 using namespace std;

 class Bank {
 public:
     static int coinsInBank = 10000;
     static int distribute(int numberOfCoinsRequested) {
         int numberOfCoinsToVend = min(numberOfCoinsRequested, coinsInBank);
         coinsInBank -= numberOfCoinsToVend;
         return numberOfCoinsToVend;
     }

     static void receive(int coins) {
         coinsInBank += coins;
     }
};

class Player {
    private:
        int coinsInPurse;
    public:
        Player(int coins){
            coinsInPurse = Bank::distribute(coins);
        }

        void win(int coins) {
            coinsInPurse += Bank::distribute(coins);
        }

        ~Player() {
            Bank::receive(coinsInPurse);
        }
};

// Main function
int main() {
    Player *playerOne = new Player(100);
    playerOne->win(2000);
    cout << "PlayerOne won 2000 coins & now has " << playerOne->coinsInPurse << "coins" << endl; 
    cout << "The bank now only has" << Bank::coinsInBank << "coins left" << endl;
    playerOne = nullptr;
    cout << "PlayerOne has left the game" << endl;
    cout << "The bank now has " << Bank::coinsInBank << "coins" << endl;
    return 0;
}